geode-core/src/integrationTest/java/org/apache/geode/distributed/internal/membership/gms/locator/GMSLocatorRecoveryIntegrationTest.java

```diff
@@ -153,18 +155,24 @@ public void testRecoverFromOther() throws Exception {
     DistributionConfigImpl config = new DistributionConfigImpl(nonDefault);
     RemoteTransportConfig transport = new RemoteTransportConfig(config, NORMAL_DM_TYPE);
 
-    DistributedMembershipListener mockListener = mock(DistributedMembershipListener.class);
+    MembershipListener mockListener = mock(MembershipListener.class);
+    MessageListener mockMessageListener = mock(MessageListener.class);
     InternalDistributedSystem mockSystem = mock(InternalDistributedSystem.class);
     DMStats mockDmStats = mock(DMStats.class);
 
     when(mockSystem.getConfig()).thenReturn(config);
 
     // start the membership manager
-    membershipManager = MemberFactory.newMembershipManager(mockListener, transport,
-        mockDmStats,
-        new GMSAuthenticator(mockSystem.getSecurityProperties(), mockSystem.getSecurityService(),
-            mockSystem.getSecurityLogWriter(), mockSystem.getInternalLogWriter()),
-        mockSystem.getConfig());
+    membershipManager =
+        MembershipBuilder.newMembershipBuilder(null)
+            .setAuthenticator(new GMSAuthenticator(mockSystem.getSecurityProperties(),
+                mockSystem.getSecurityService(),
+                mockSystem.getSecurityLogWriter(), mockSystem.getInternalLogWriter()))
+            .setStatistics(mockDmStats)
```

Comment:
We modified this test with a refactoring command from a long method call to a builder. However, I think right now they are all required since we are not providing any empty implementations by default. Maybe we can change that in the API to make some of these optional at a later point?

Reply:
We made this change with a refactoring command - from a long method to a builder. But right now there is no empty implementation of these objects, so they are all required. We can look at providing default empty implementations later if we want.